sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/constraints.scala

```diff
@@ -92,6 +98,11 @@ trait TableConstraint {
       )
     }
   }
+
+  override def nullable: Boolean = true
+
+  override def dataType: DataType =
+    throw new SparkUnsupportedOperationException("CONSTRAINT_DOES_NOT_HAVE_DATA_TYPE")
```

Comment:
Hmm, by default `dataType` is not supported, but it has default `nullable`? Does it make sense?

Reply:
Can you explain what is the issue here?

Reply:
I'm not sure. Is there a case you will use nullable without dataType? Should it be unsupported too by default for nullable?

Not a big deal, though. Just wondering the reason.

Reply:
Either way should be ok.  I changed to throwing exception.